src/clientiface.h

```diff
@@ -243,15 +243,18 @@ class RemoteClient
 	AuthMechanism chosen_mech  = AUTH_MECHANISM_NONE;
 	void *auth_data = nullptr;
 	u32 allowed_auth_mechs = 0;
-	u32 allowed_sudo_mechs = 0;
 
 	void resetChosenMech();
 
-	bool isSudoMechAllowed(AuthMechanism mech)
-	{ return allowed_sudo_mechs & mech; }
 	bool isMechAllowed(AuthMechanism mech)
 	{ return allowed_auth_mechs & mech; }
 
+	void setEncryptedPassword(const std::string& pwd) {
+		enc_pwd = pwd;
+		// We just set SRP encrypted password, we accept only it now
```

Comment:
suggestion: `assert(pwd.find("#1#") == 0); // must be srp`

Reply:
Makes sense, done